service/history/api/startworkflow/api.go

```diff
@@ -506,17 +507,44 @@ func (s *Starter) respondToRetriedRequest(
 	return s.generateResponse(runID, mutableStateInfo.workflowTask, events)
 }
 
+func (s *Starter) getWorkflowStartTime(ctx context.Context, runID string) (start time.Time, retError error) {
+	mutableState, releaseFn, retError := s.getMutableState(ctx, runID)
+	var workflowStartTime time.Time
+
+	defer func() {
+		// do we need "recover" here?
+		releaseFn(retError)
+	}()
+
+	if retError != nil {
+		return workflowStartTime, retError
+	}
+	if mutableState == nil {
+		return workflowStartTime, serviceerror.NewInternal("Can't load mutable state")
```

Comment:
I don't think this should happen 🤔 I'd expect there to be an error (now).

Reply:
I also think so, but will not hurt I guess

Reply:
I would rather remove this line. If all our code start paranoid like this we would have much harder code to read.